### ISSUE TITLE
Variational Posterior: precision parameters are not updating

### ISSUE THREAD

**Issue by mbronckers** [mbronckers/GI-PVI#2]
**Bug**: the `nz` variational precision parameters are not changing across optimization.

**Reason**: 
- inside Varz, we initialize a `positive` variable which applies a transformation to the original variable and removes the requires_grad.
- inside our `ts`, we only refer to the original variable, which means gradients are not being applied to the ts.

**Possible solution**:
1. Build the `ts` from scratch / update the `ts` before each `estimate_elbo` call.

**Comment by mbronckers** [comment]
- Implemented solution 1, but no luck so far. 

- The `requires_grad` is now set to `True` for the `prec` passed to `NaturalNormal`, but I believe we lose the gradients when we convert it to an AbstractMatrix:
```
class NaturalNormal:
    def __init__(self, lam, prec):
        """
        :param lam: first natural parameter of Normal dist = precision x mean
        :param prec: second natural parameter of Normal dist = -0.5 x precision \\propto precision 
        """
        self.lam = lam 
        self.prec = convert(prec, AbstractMatrix)
        
        self._mean = None
        self._var = None
```
<img width="555" alt="image" src="https://user-images.githubusercontent.com/25270093/171384904-a2513a25-56ac-4f40-9ffa-51844f4dd3cc.png">

**Comment by mbronckers** [comment]
**Rebuild()** solves the gradient issue
